benchmarks/src/main/scala/benchmark/MergeTestCases.scala

```diff
@@ -0,0 +1,251 @@
+    rowNotMatchedFraction: Double) extends InsertOnlyTestCase {
+
+  override val name: String = "multiple_insert_only" +
+    s"_fileMatchedFraction_$fileMatchedFraction" +
+    s"_rowNotMatchedFraction_$rowNotMatchedFraction"
+
+  override def sqlCmd(targetTable: String): String = {
+    s"""MERGE INTO $targetTable t
+        |USING ${sourceTable.name} s
+        |ON t.wr_order_number = s.wr_order_number AND t.wr_item_sk = s.wr_item_sk
+        |WHEN NOT MATCHED AND s.wr_item_sk % 2 = 0 THEN INSERT *
+        |WHEN NOT MATCHED THEN INSERT *""".stripMargin
+   }
+}
+
+/**
+ * A merge test case with a single WHEN MATCHED THEN DELETED clause.
+ */
+case class DeleteOnlyTestCase(
+    fileMatchedFraction: Double,
+    rowMatchedFraction: Double) extends MergeTestCase {
+
+  override val name: String = "delete_only" +
+    s"_fileMatchedFraction_$fileMatchedFraction" +
+    s"_rowMatchedFraction_$rowMatchedFraction"
+
+  override def sourceTable: MergeSourceTable = MergeSourceTable(
+    fileMatchedFraction,
+    rowMatchedFraction,
+    rowNotMatchedFraction = 0)
+
+  override def sqlCmd(targetTable: String): String = {
+    s"""MERGE INTO $targetTable t
+        |USING ${sourceTable.name} s
+        |ON t.wr_order_number = s.wr_order_number AND t.wr_item_sk = s.wr_item_sk
+        |WHEN MATCHED THEN DELETE""".stripMargin
+   }
+
+  override def validate(mergeStats: Seq[Row], targetRowCount: Long): Unit = {
+    assert(mergeStats.length == 1)
+    assert(mergeStats.head.getAs[Long]("num_updated_rows") == 0)
+    assert(mergeStats.head.getAs[Long]("num_inserted_rows") == 0)
+  }
+}
+
+/**
+ * A merge test case with a MATCHED UPDATE and a NOT MATCHED INSERT clause.
+ */
+case class UpsertTestCase(
+    fileMatchedFraction: Double,
+    rowMatchedFraction: Double,
+    rowNotMatchedFraction: Double) extends MergeTestCase {
+
+  override val name: String = "upsert" +
+    s"_fileMatchedFraction_$fileMatchedFraction" +
+    s"_rowMatchedFraction_$rowMatchedFraction" +
+    s"_rowNotMatchedFraction_$rowNotMatchedFraction"
+
+  override def sourceTable: MergeSourceTable = MergeSourceTable(
+    fileMatchedFraction,
+    rowMatchedFraction,
+    rowNotMatchedFraction)
+
+  override def sqlCmd(targetTable: String): String = {
+    s"""MERGE INTO $targetTable t
+        |USING ${sourceTable.name} s
+        |ON t.wr_order_number = s.wr_order_number AND t.wr_item_sk = s.wr_item_sk
+        |WHEN MATCHED THEN UPDATE SET *
+        |WHEN NOT MATCHED THEN INSERT *""".stripMargin
+   }
+
+  override def validate(mergeStats: Seq[Row], targetRowCount: Long): Unit = {
+    assert(mergeStats.length == 1)
+    assert(mergeStats.head.getAs[Long]("num_deleted_rows") == 0)
+  }
+}
+
+object MergeTestCases {
+  def testCases: Seq[MergeTestCase] =
+    insertOnlyTestCases ++
+    deleteOnlyTestCases ++
+    upsertTestCases
+
+  def insertOnlyTestCases: Seq[MergeTestCase] = Seq(
+    SingleInsertOnlyTestCase(
+      fileMatchedFraction = 0.05,
```

Comment:
Maybe add a fileMatchedFractionSeq, rowNotMatchedFractionSeq and do a loop? Same for the tests below.

Reply:
I've grouped test definitions when it makes sense